src/utils/time_utils.cpp

```diff
@@ -38,50 +60,65 @@ std::string time_utils::timepoint_to_readable(const time_utils::t_timepoint &tp,
 
 std::string time_utils::time_t_to_readable(const std::time_t &time, const std::string &zone) {
 
+	// Pacific/Marquesas and America/St_Johns warn
+	// This two zones might not works properly.
+	// However it looks like boost::date_time fixed this
+	if(zone == "Pacific/Marquesas" || zone == "America/St_Johns") {
+		_warn("Time for " << zone << " zone could not works properly!");
+	}
+
 	std::string full_date = get_date_str(time);
 
 	if(zone.empty()) {
-		full_date += time_utils::get_zone_offset_local(time);
+		full_date += time_utils::get_utc_offset_string();
 	} else {
-		full_date += time_utils::get_zone_offset_universal(time, zone);
+		full_date += time_utils::get_zone_utc_offset(zone);
 	}
 
 	return full_date;
 }
 
-// use of std::strftime instead of std::put_time, because of no implementation in gcc version < 5 (4.9.2 on debian8)
 std::string time_utils::get_date_str(const std::time_t &time) {
-	char buff[35];
-	std::strftime(buff, sizeof(buff), "%FT%T", std::gmtime(&time));
+	using namespace boost::posix_time;
+	ptime  t = from_time_t(time);
 
-	return std::string(buff);
+	return to_iso_extended_string(t) ;
 }
 
-// use of std::strftime instead of std::put_time, because of no implementation in gcc version < 5 (4.9.2 on debian8)
-std::string time_utils::get_zone_offset_local(const std::time_t &time) {
+boost::posix_time::time_duration time_utils::get_utc_offset() {
```

Comment:
pass the timepoint here,   we will return zone diff at this point in time
(because some zones have varaible offset - due to e.g. summertime / lightsaving)

https://stackoverflow.com/questions/8746848/boost-get-the-current-local-date-time-with-current-time-zone-from-the-machine/8750438#8750438

e.g. like code

boost::posix_time::time_duration getUtcOffset(const boost::posix_time::ptime& utcTime)
{
    using boost::posix_time::ptime;
    const ptime localTime = boost::date_time::c_local_adjustor<ptime>::utc_to_local(utcTime);
    return localTime - utcTime;
}